ansys_optical_automation/application/example_create_material_file_from_decadic_absorbance_data.py

```diff
@@ -0,0 +1,217 @@
+    return line_float_list
+
+
+def select_datafile():
+    """
+    Function that opens a dialog window for the user to select the file containing the decadic absorbance data. This
+    file is in principle a txt, xls, xlsx or dat file. The file content must consist of two columns. The first column
+    contains the wavenumber values (cm^-1) and the second column contains the corresponding decadic absorbance values
+    (1 ppm-meter).
+
+    Returns
+    -------
+    A string of the full path of the selected file.
+    """
+    root = tk.Tk()
+    root.withdraw()
+    file_path = filedialog.askopenfilename(title="Select file containing decadic absorbance data")
+    return file_path
+
+
+def calculate_transmittance_from_dec_absorbance(dec_absorbance: float):
+    """
+    Function to calculate the transmittance value from the decadic absorbance value of a material at 1ppm per one meter
+    of material.
+
+    Parameters
+    ----------
+    dec_absorbance : A float representing the decadic absorbance of a material (usually at a specific wavelength).
+    The decadic absorbance is physically defined as dec_absorbance=I(0)/I(l), where I(0) is the incident light and I(l)
+    absorbance is 10^dec_absorbance = exp(absorbance).
+
+    Returns
+    -------
+    A float representing the transmittance of a material (usually at a specific wavelength). The transmittance
+    is physically defined as transmittance=I(l)/I(0).
+    """
+    transmittance = 10 ** (-dec_absorbance)  # [ - ] (ppm-meter@296K)
+    return transmittance
+
+
+def calculate_absorption_coef_from_dec_absorbance(dec_absorbance: float):
+    """
+    Function that calculates the absorption coefficient of a material from the decadic absorbance value of a material at
+    1ppm per one meter of material. The absorption coefficient is calculated from the Beer-Lambert law
+    I(l)=I(0)exp(-absorption_coef*l) <=> absorption_coef=-ln(transmittance)/l, where I(0) is the incident light and I(l)
+    the light transmitted through the propagation distance l.
+
+    Parameters
+    ----------
+    dec_absorbance : A float representing the decadic absorbance of a material (usually at a specific wavelength).
+    The decadic absorbance is physically defined as dec_absorbance=I(0)/I(l).
+
+    Returns
+    -------
+    A float representing the absorption coefficient (mm^-1) of a material (usually at a specific wavelength).
+    NOTE: The relation between absorption coefficient and decadic absorption coefficient is
+    dec_absorption_coef=absorption_coef/ln(10).
+    """
+    transmittance = calculate_transmittance_from_dec_absorbance(dec_absorbance)
+    absorption_coef = -math.log(transmittance) / 1e3  # [mm^-1]
+    return absorption_coef
+
+
+def calculate_wavelength_from_wavenumber(wavenumber: float):
+    """
+    Function that calculates the wavelength (nm) from the wavenumber (cm^-1).
+
+    Parameters
+    ----------
+    wavenumber : A float representing the wavenumber value (cm^-1)
```

Comment:
@ssklaven you need to provide the type of parameters after the parameter and description as the second line :). Example can be found: https://numpydoc.readthedocs.io/en/latest/format.html